File: src/main/java/demo/AppRoutes.java
package demo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.MediaType;
import org.springframework.web.reactive.function.server.RouterFunction;
import sample.handler.MessageHandler;

import static org.springframework.web.reactive.function.server.RequestPredicates.*;
import static org.springframework.web.reactive.function.server.RouterFunctions.nest;
import static org.springframework.web.reactive.function.server.RouterFunctions.route;

@Configuration
public class AppRoutes {

    @Autowired
    private MessageHandler messageHandler;

    @Bean
    RouterFunction<?> apis() {
        return nest(path("/hotels"), nest(accept(MediaType.APPLICATION_JSON),
                route(
                        GET("/"), messageHandler::getMessages)
                        .andRoute(POST("/"), messageHandler::addMessage)
                        .andRoute(GET("/{id}"), messageHandler::getMessage)
                        .andRoute(PUT("/{id}"), messageHandler::updateMessage)
                        .andRoute(DELETE("/{id}"), messageHandler::deleteMessage)
        ));
    }

}

File: src/main/kotlin/sample/routes/AppRoutes.kt
package sample.routes

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.MediaType.APPLICATION_JSON
import org.springframework.web.reactive.function.server.router
import sample.handler.MessageHandler

@Configuration
class AppRoutes(private val messageHandler: MessageHandler) {

    @Bean
    fun apis() = router {
        (accept(APPLICATION_JSON) and "/messages").nest {
            GET("/", messageHandler::getMessages)
            POST("/", messageHandler::addMessage)
            GET("/{id}", messageHandler::getMessage)
            PUT("/{id}", messageHandler::updateMessage)
            DELETE("/{id}", messageHandler::deleteMessage)
        }
    }

}